### PR TITLE
Fix uninitialized variable causing `goagen client --notool` crash

### PR DESCRIPTION
**Pull request by Syeberman**

### CODE REVIEW

File: goagen/gen_client/generator_test.go [hunk 1]
```diff
@@ -90,6 +90,17 @@ var _ = Describe("Generate", func() {
...
+		Context("with --notool", func() {
+			BeforeEach(func() {
+				os.Args = append(os.Args, "--notool")
+			})
+
+			It("should not return an error", func() {
+				Ω(genErr).Should(BeNil())
+				Ω(files).Should(HaveLen(5)) // 9, minus 4 entries for tool paths
+			})
+		})
```
Comment (Syeberman) [inline goagen/gen_client/generator_test.go]: If you run commit 8e2e4d5 on its own you'll see this test crash.

File: goagen/gen_client/generator.go [hunk 2]
```diff
@@ -149,14 +149,16 @@ func (g *Generator) Generate() (_ []string, err error) {
...
 		}
+		arrayToStringTmpl = template.Must(template.New("client").Funcs(funcs).Parse(arrayToStringT))
+	}
+
+	if !g.NoTool {
+		var cliPkg string
 		cliPkg, err = codegen.PackagePath(cliDir)
 		if err != nil {
 			return
 		}
```
Comment (Syeberman) [inline goagen/gen_client/generator.go]: GitHub's diff doesn't make this clear, but all I've done is moved the above `cliPkg` code down to inside the `if !g.NoTool` block.  `cliDir` is only set when `g.NoTool` is false, so using `--notool` would always result in a panic.

**Comment by raphael** [comment]
Thank you for the fix and clear PR! Do you think you could backport the fix to the `v1` branch? Thank you!

**Comment by Syeberman** [comment]
Check out https://github.com/goadesign/goa/pull/882 for the v1 backport

**Pull request merged**